test/clj_kondo/analysis/java_test.clj

```diff
@@ -105,12 +105,53 @@
         :filename #".*corpus/java/sources/foo/bar/AwesomeClass.java"}]
      java-class-definitions))
   (testing "linting just one java source"
-    (let [{:keys [java-class-definitions]} (analyze ["corpus/java/sources/foo/bar/AwesomeClass.java"])]
+    (let [{:keys [java-class-definitions java-member-definitions]} (analyze ["corpus/java/sources/foo/bar/AwesomeClass.java"])]
       (assert-submaps2
        '[{:class "foo.bar.AwesomeClass",
           :uri #"file:.*/corpus/java/sources/foo/bar/AwesomeClass.java",
           :filename #".*corpus/java/sources/foo/bar/AwesomeClass.java"}]
-       java-class-definitions))))
+       java-class-definitions)
+      (assert-submaps2
+       '[{:class "foo.bar.AwesomeClass"
+          :uri #"file:.*/corpus/java/sources/foo/bar/AwesomeClass.java"
+          :flags #{:public :field}
+          :name "bar1"
+          :type "Double"
+          :row 12 :col 5 :end-row 12 :end-col 23}
+         {:class "foo.bar.AwesomeClass"
+          :uri #"file:.*/corpus/java/sources/foo/bar/AwesomeClass.java"
+          :flags #{:public :field :final}
+          :name "bar2"
+          :type "Double"
+          :row 13 :col 5 :end-row 13 :end-col 35}
+         {:class "foo.bar.AwesomeClass"
+          :uri #"file:.*/corpus/java/sources/foo/bar/AwesomeClass.java"
+          :flags #{:public :static :field :final}
+          :name "bar3"
+          :type "Double"
```

Comment:
Should this type be fully qualified, as it can refer to any class?

```
java.lang.Double
```

Reply:
Yeah, I'm not sure if java parser does that only for java.langs, but I couldn't think on how to get the full package there, any suggestions?

Reply:
For that to work, javaparser need to resolve the symbol which is possible via its javaparser-symbol-solver lib, now it's properly working!